enhancements/metering/user-provided-pricesheets.md

```diff
@@ -0,0 +1,205 @@
+
+## Proposal
+
+Currently, cost calculations for AWS are done by determining the total cluster cost for the reporting period by looking at the AWS billing data, using node metrics to determine the nodes active during the reporting period.
+This total cluster cost was then split up by pod, or namespace by calculating the percentage of usage or request for the pod/namespaces in terms of CPU or memory.
+This percentage was then multiplied by the cost for the reporting period to get the total cost for a pod or namespace.
+
+The price sheet model changes how we calculate cost, but actually provides more granularity and actually translates to the same concepts as above but with a bit more complexity as a result of the flexibility.
+
+Calculating cost with a price sheet is about having a price for a rate of usage and multiplying the usage by that price, much like the way we do cost reporting with AWS today.
+The primary difference is that instead of doing this over the entire reporting period, we do it per pricing period in the billing report, and the calculation is done by calculating the price per pricing period and summing the results of that over the entire reporting period.
+
+We start by enabling users to define price sheets by documenting how to do so with existing functionality that's available, and by defining a few more default ReportQueries that utilize price sheets when calculating costs.
+
+At a high level this means we need to do the following:
+
+- Define a Presto table schema that supports both user-provided prices and calculated prices from cloud billing data.
+- Create built in [ReportQuery][reportqueries] and [ReportDataSource][reportdatasources] that can create price sheet table using AWS billing data.
+- Create built in [ReportQueries][reportqueries] that calculate costs using a price sheet specified as a [Report input][report-input].
+
+### User Stories [optional]
+
+- As a member of the operations team, I want to create a Report that let's me calculate costs with any price sheet.
+- As a member of the operations team, I want to be able to define my own prices to be used with Reports.
+
+### Implementation Details/Notes/Constraints [optional]
+
+
+#### Price Sheet Table Schema
+
+The Presto table schema is defining a set of column names and their data types that all price sheet Presto database tables will have, enabling ReportQueries designed to use price sheets to work with any of them interchangeably.
+
+The price sheet will define 7 columns.
+
+- `resource varchar` is the type of resource, such as CPU or Memory
+- `resource_unit varchar` is the unit of the resource. For CPU it might be cores or millicores and for memory it could be bytes, gigabytes, etc.
+- `time_unit varchar` is the unit of time that the resource is billed at. For example 'hour' or 'minute'.
```

Comment:
Should this be `enum` to limit potential for bad data?

Reply:
Unfortunately we don't have enums really.